com.unity.netcode.gameobjects/Runtime/Core/NetworkObject.cs

```diff
@@ -969,20 +969,21 @@ public bool TrySetParent(NetworkObject parent, bool worldPositionStays = true)
                 return false;
             }
 
-            if (!NetworkManager.IsServer)
+            if (!NetworkManager.IsServer && !NetworkManager.ShutdownInProgress)
             {
                 return false;
             }
 
-            if (!IsSpawned)
+            // If the parent is not null fail only if either of the two is true:
+            // - This instance is spawned and the parent is not.
+            // - This instance is not spawned and the parent is.
+            // Basically, don't allow parenting when either the child or parent is not spawned.
+            // Caveat: if the parent is null then we can allow parenting whether the instance is or is not spawned.
+            if (parent != null && (IsSpawned && !parent.IsSpawned || !IsSpawned && parent.IsSpawned))
```

Comment:
Totally optional comment, BUT... `(IsSpawned && !parent.IsSpawned || !IsSpawned && parent.IsSpawned)` could be simplified to `IsSpawned ^ parent.IsSpawned`.